js/ui/components/clubhouse.js

```diff
@@ -0,0 +1,323 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+//
+// Copyright (C) 2018 Endless Mobile, Inc.
+//
+// Licensed under the GNU General Public License Version 2
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// Author: Joaquim Rocha <jrocha@endlessm.com>
+//
+
+const Clutter = imports.gi.Clutter;
+const Flatpak = imports.gi.Flatpak
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const Shell = imports.gi.Shell;
+const Signals = imports.signals;
+const St = imports.gi.St;
+
+const GnomeSession = imports.misc.gnomeSession;
+const Lang = imports.lang;
+const Main = imports.ui.main;
+const MessageList = imports.ui.messageList;
+const MessageTray = imports.ui.messageTray;
+const Monitor = imports.ui.monitor;
+const NotificationDaemon = imports.ui.notificationDaemon;
+
+const sessionBus = Gio.bus_get_sync(Gio.BusType.SESSION, null);
+
+const CLUBHOUSE_ID = 'com.endlessm.Clubhouse';
+const CLUBHOUSE_DBUS_OBJ_PATH = '/com/endlessm/Clubhouse';
+
+var ClubhouseNotificationBanner = new Lang.Class({
+    Name: 'ClubhouseNotificationBanner',
+    Extends: MessageTray.NotificationBanner,
+
+    _init: function(notification) {
+        this.parent(notification);
+
+        // We don't have an "unexpanded" state for now
+        this.expanded = true;
+        this._actionBin.visible = true;
+
+        this.actor.can_focus = true;
+        this.actor.track_hover = true;
+
+        // Override the style name because this is a not a regular notification
+        this.actor.remove_style_class_name('notification-banner');
```

Comment:
The idea was to clear any notion of the `notification-banner`. Why should we keep the old style?

Reply:
I am sure it can be made to work either way, but it's a bit unusual to remove style classes, since it can cause unexpected issues when e.g. selectors exist that target a child through a parent->child chain.
Anyway, you can keep it like this if you prefer it this way.

Reply:
I prefer to keep it because in principle our notifications should have anything to do with the default ones in terms of styling.